lectures/textures/SDLProject/main.cpp

```diff
@@ -0,0 +1,288 @@
+#define GL_SILENCE_DEPRECATION
+#define STB_IMAGE_IMPLEMENTATION
+
+#ifdef _WINDOWS
+#include <GL/glew.h>
+#endif
+
+#define GL_GLEXT_PROTOTYPES 1
+#include <SDL.h>
+#include <SDL_opengl.h>
+#include "glm/mat4x4.hpp"
+#include "glm/gtc/matrix_transform.hpp"
+#include "ShaderProgram.h"
+#include "stb_image.h"
+
+enum Coordinate
+{
+    x_coordinate,
+    y_coordinate
+};
+
+enum AppStatus { RUNNING, TERMINATED };
+
+#define LOG(argument) std::cout << argument << '\n'
+
+constexpr int WINDOW_WIDTH = 640,
+          WINDOW_HEIGHT = 480;
+
+constexpr float BG_RED = 0.1922f,
+            BG_BLUE = 0.549f,
+            BG_GREEN = 0.9059f,
+            BG_OPACITY = 1.0f;
+
+constexpr int VIEWPORT_X = 0,
+          VIEWPORT_Y = 0,
+          VIEWPORT_WIDTH = WINDOW_WIDTH,
+          VIEWPORT_HEIGHT = WINDOW_HEIGHT;
+
+constexpr char V_SHADER_PATH[] = "shaders/vertex_textured.glsl",
+           F_SHADER_PATH[] = "shaders/fragment_textured.glsl";
+
+constexpr float MILLISECONDS_IN_SECOND = 1000.0;
+constexpr float DEGREES_PER_SECOND = 90.0f;
+
+constexpr int NUMBER_OF_TEXTURES = 1; // to be generated, that is
+constexpr GLint LEVEL_OF_DETAIL = 0;  // base image level; Level n is the nth mipmap reduction image
+constexpr GLint TEXTURE_BORDER = 0;   // this value MUST be zero
+
+constexpr char PLAYER_SPRITE_FILEPATH[] = "soph.png";
+
+AppStatus g_app_status = RUNNING;
+
+SDL_Window* g_display_window;
+bool g_game_is_running = true;
+bool g_is_growing = true;
+
+ShaderProgram g_shader_program = ShaderProgram();
+glm::mat4 g_view_matrix,
+          g_model_matrix,
+          g_projection_matrix,
+          g_trans_matrix;
+
+float g_previous_ticks = 0.0f;
+
+GLuint g_player_texture_id;
+SDL_Joystick *g_player_one_controller;
+
+// overall position
+glm::vec3 g_player_position = glm::vec3(0.0f, 0.0f, 0.0f);
+
+// movement tracker
+glm::vec3 g_player_movement = glm::vec3(0.0f, 0.0f, 0.0f);
+
+float get_screen_to_ortho(float coordinate, Coordinate axis)
+{
+    switch (axis) {
+        case x_coordinate:
+            return ((coordinate / WINDOW_WIDTH) * 10.0f ) - (10.0f / 2.0f);
+        case y_coordinate:
+            return (((WINDOW_HEIGHT - coordinate) / WINDOW_HEIGHT) * 7.5f) - (7.5f / 2.0);
+        default:
+            return 0.0f;
+    }
+}
```

Comment:
I'm not sure why this is here (it was here last semester too), but this is from next week's lecture, so you can get rid of it